app/Commands/NewCommand.php

```diff
@@ -0,0 +1,474 @@
+        }
+
+        if ($this->option('node')) {
+            $this->doNodeOrYarn();
+        }
+
+        if ($this->option('link')) {
+            if ($this->tools['valet']) {
+                if ($this->doValetLink()) {
+                    $this->openBrowser();
+                }
+            }
+        }
+
+        if ($this->tools['git']) {
+            $this->doGit();
+        }
+
+        $this->openTextEditor();
+
+        $this->info("You're ready to go! Remember to cd into '{$this->projectpath}' before you start editing.");
+    }
+
+    protected function getAvailableTools()
+    {
+        $finder = new ExecutableFinder();
+
+        $checks = ['yarn', 'npm', 'git', 'valet', 'laravel', 'composer'];
+        foreach ($checks as $check) {
+            $this->tools[$check] = $finder->find($check);
+        }
+    }
+
+    protected function setBasePath()
+    {
+        $this->basepath = $this->cwd;
+        if ($this->option('path')) {
+            $path = $this->option('path');
+            if (is_dir($path)) {
+                $this->basepath = $path;
+            } else {
+                $this->warn("Your defined '--path $path' is not a directory, so I am skipping it and using '{$this->basepath}' instead.");
+            }
+        }
+
+        return $this->basepath;
+    }
+
+    protected function askToAndRemoveProject()
+    {
+        $this->info("The directory '{$this->projectpath}' already exists.");
+
+        if ($this->confirm("Shall I proceed by removing the following directory? {$this->projectpath}")) {
+            $fs = new Filesystem\Filesystem();
+
+            // @todo Use laravel --force here instead of deleteDirectory()?
+            if ($fs->deleteDirectory($this->projectpath)) {
+                $this->info("I removed the following directory: {$this->projectpath}");
+
+                return true;
+            } else {
+                $this->error("I was unable to remove the '{$this->projectpath}' directory so I must exit.");
+
+                return false;
+            }
+        } else {
+            $this->error("You have chosen to not remove the '{$this->projectpath}' directory so I must exit.");
+
+            return false;
+        }
+    }
+
+    protected function replaceEnvVariables()
+    {
+        // @todo make this a configuration option
+        $changes = [
+            'DB_DATABASE' => $this->projectname,
+            'DB_USERNAME' => 'root',
+            'DB_PASSWORD' => '',
+            'APP_URL' => $this->projecturl,
+        ];
+
+        // @todo There has to be a better way; opening/closing a file 4x is wrong
+        foreach ($changes as $name => $newvalue) {
+            $this->replaceEnvVariable($name, $newvalue);
+        }
+    }
+
+    protected function replaceEnvVariable($name, $newvalue)
+    {
+        // @todo will .env always exist here? Check if not and copy over .env.example?
+        // @todo research a more official way to replace .env values... I could not find one
```

Comment:
This one is pretty brute force but we may be able to work from this:

http://laravel-tricks.com/tricks/change-the-env-dynamically

Reply:
Thanks, fixed via commit f8d54d6. That works okay although has limitations (e.g., does not preserve comments or whitespace) so I essentially created my own. As a bonus it'll also append new key=value pairs to `.env`.